src/panel/PanelManager.jsx

```diff
@@ -214,27 +216,41 @@ export default class PanelManager extends React.Component {
   }
 
   render() {
-    const { ActivePanel, options, panelSize } = this.state;
+    const { ActivePanel, options, panelSize, isPanelVisible } = this.state;
+    const { searchBarInputNode, searchBarOutputNode } = this.props;
 
     return <div>
       <Suggest
-        inputNode={this.props.searchBarInputNode}
-        outputNode={document.querySelector('.search_form__result')}
+        inputNode={searchBarInputNode}
+        outputNode={searchBarOutputNode}
         withCategories
         onChange={query => {
           if (ActivePanel === ServicePanel) {
-            togglePanelVisibility(query.length === 0);
-          } else {
-            togglePanelVisibility(false);
+            this.setState({ isPanelVisible: query.length === 0 });
+          } else if (isPanelVisible) {
```

Comment:
Do we really need all these explicit checks on the previous `isPanelVisible` value?
Unless there is a blatant perf problem or side effect, it would be easier to read and understand if you only consider the condition on the panel and let React do its job with state updates.
```js
this.setState({ isPanelVisible: ActivePanel === ServicePanel && query.length === 0 });
```
Same for the other callbacks.

Reply:
If we encounter perf problems with `setState` calls that trigger too many re-renders, one way to solve it is to switch to a functional component, so state comparisons are on atomic values and won't trigger useless renders.

Reply:
yes it should works, I was not really focus on optimizing things, instead I tried to make things work at first :zany_face: 
Thank you